src/seedsigner/helpers/embit_utils.py

```diff
@@ -93,3 +98,26 @@ def get_multisig_address(descriptor: Descriptor, index: int = 0, is_change: bool
     elif descriptor.is_taproot:
         # TODO: Not yet implemented!
         raise Exception("Taproot verification not yet implemented!")
+
+
+
+def sign_message(seed_bytes: bytes, derivation, msg: bytes, compressed: bool = True, embit_network: str="main") -> bytes:
+    """
+        from: https://github.com/cryptoadvance/specter-diy/blob/b58a819ef09b2bca880a82c7e122618944355118/src/apps/signmessage/signmessage.py
+    """
+    """Sign message with private key"""
+    msghash = sha256(
+        sha256(
+            b"\x18Bitcoin Signed Message:\n" + compact.to_bytes(len(msg)) + msg
+        ).digest()
+    ).digest()
+
+    root = bip32.HDKey.from_seed(seed_bytes, version=NETWORKS[embit_network]["xprv"])
+    prv = root.derive(derivation).key
+    sig = secp256k1.ecdsa_sign_recoverable(msghash, prv._secret)
+    flag = sig[64]
```

Comment:
confusing variable name --> first byte of what `secp256k1.ecdsa_sign_recoverable` returns is recovery id --> `rec_id = sig[64]`